modules/cms/classes/Controller.php

```diff
@@ -1408,32 +1408,54 @@ public function param($name, $default = null)
      * @param string $alias       Alias to give the component
      * @param array  $properties  Component properties
      * @param bool   $addToLayout Add to layout, instead of page
-     * @return ComponentBase Component object
+     * @return ComponentBase|null Component object. Will return `null` if a soft component is used but not found.
+     * @throws CmsException
+     * @throws SystemException
      */
     public function addComponent($name, $alias, $properties, $addToLayout = false)
     {
         $manager = ComponentManager::instance();
 
-        if ($addToLayout) {
-            if (!$componentObj = $manager->makeComponent($name, $this->layoutObj, $properties)) {
-                throw new CmsException(Lang::get('cms::lang.component.not_found', ['name'=>$name]));
-            }
-
-            $componentObj->alias = $alias;
-            $this->vars[$alias] = $this->layout->components[$alias] = $componentObj;
+        $isSoftComponent = $this->isSoftComponent($name);
+        if ($isSoftComponent) {
+            $name = $this->parseComponentLabel($name);
+            $alias = $this->parseComponentLabel($alias);
         }
-        else {
-            if (!$componentObj = $manager->makeComponent($name, $this->pageObj, $properties)) {
-                throw new CmsException(Lang::get('cms::lang.component.not_found', ['name'=>$name]));
+        try {
+            if ($addToLayout) {
+                if (!$componentObj = $manager->makeComponent($name, $this->layoutObj, $properties, $isSoftComponent)) {
+                    throw new CmsException(Lang::get('cms::lang.component.not_found', ['name' => $name]));
+                }
+
+                $componentObj->alias = $alias;
+                $this->vars[$alias] = $this->layout->components[$alias] = $componentObj;
+            } else {
+                if (!$componentObj = $manager->makeComponent($name, $this->pageObj, $properties, $isSoftComponent)) {
+                    throw new CmsException(Lang::get('cms::lang.component.not_found', ['name' => $name]));
+                }
+
+                $componentObj->alias = $alias;
+                $this->vars[$alias] = $this->page->components[$alias] = $componentObj;
             }
 
-            $componentObj->alias = $alias;
-            $this->vars[$alias] = $this->page->components[$alias] = $componentObj;
+            $this->setComponentPropertiesFromParams($componentObj);
+            $componentObj->init();
+
+            return $componentObj;
+        /**
+         * We catch two exceptions below, as missing plugin can throw both and catch \Exception would be too global.
+         */
```

Comment:
Just make this a single line comment `//` and add an extra line of whitespace between return and the comment